zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -177,13 +332,58 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
             &mut reversion_info,
         );
 
-        // TODO: Handle creation transaction
         // TODO: Handle precompiled
 
-        let is_empty_code_hash =
-            IsEqualGadget::construct(cb, phase2_code_hash.expr(), cb.empty_hash_rlc());
-        let is_zero_code_hash = IsZeroGadget::construct(cb, phase2_code_hash.expr());
-        let is_empty_code = or::expr([is_empty_code_hash.expr(), is_zero_code_hash.expr()]);
+        // Handle contract creation transaction
+        // tx_is_create = true when tx.to = none
+        cb.condition(tx_is_create.expr(), |cb| {
```

Comment:
i read again the upstream codes, and i realized that in their `cb.condition(not::expr(is_empty_hash), ... )`, the tx.create = true case is also handled there.   So we can choose either (A) like this PR, add a "cb.condition(tx_is_create.expr()" branch, or (B) like upstream, handle tx_is_create inside "not::expr(is_empty_hash)" branch.  Both seems ok for me.